### PR TITLE
Pacbio its

### PR DESCRIPTION
**Pull request by jtangrot**
Add support for PacBio data, so far only tested on ITS data:
* Add ability to read single-end sequence data using manifest file
* Add process for generating ASVs from PacBio data using DADA2 directly
* Add ability to read and generate qiime classifier from UNITE fasta file

## PR checklist
 - [x] This comment contains a description of changes (with reason)
 - [ ] If you've fixed a bug or added code that should be tested, add tests!
 - [ ] If necessary, also make a PR on the [nf-core/ampliseq branch on the nf-core/test-datasets repo]( https://github.com/nf-core/test-datasets/pull/new/nf-core/ampliseq)
 - [ ] Ensure the test suite passes (`nextflow run . -profile test,docker`).
 - [ ] Make sure your code lints (`nf-core lint .`).
 - [x] Documentation in `docs` is updated
 - [ ] `CHANGELOG.md` is updated
 - [ ] `README.md` is updated

### CODE REVIEW

**Comment by d4straub** [comment]
Great, thanks for the PR! I'm going to have a more detailed look later.

What seems missing from a first screening is:

- an update of `CHANGELOG.md` and 
- a description of `docs/output.md` (or are there no changes to where and what files are produced?)
- test datasets

I think it would be really helpful (even for reviewing this PR) to add a small test dataset for PacBio reads and a small ITS database (see [here](https://github.com/nf-core/test-datasets/blob/master/docs/ADD_NEW_DATA.md)), so that tests can be added for those two, e.g. `conf/test_pacbio.config`, `conf/test_pacbio_its.config`, and eventually also `conf/test_illumina_its.config`.

File: main.nf [hunk 1]
```diff
@@ -602,7 +638,44 @@ if (!params.Q2imported){
...
+				"""
+			} else {
+				"""
+				qiime tools import \
+					--type 'SampleData[SequencesWithQuality]' \
+					--input-path ${manifest} \
+					--output-path demux.qza \
+					--input-format SingleEndFastqManifestPhred64
 				"""
+			}
```
Comment (d4straub) [inline main.nf:678]: I suggest simplifying:

```suggestion
			script:
			input_format = params.phred64 ? "PairedEndFastqManifestPhred64" : "PairedEndFastqManifestPhred33"
			"""
			qiime tools import \
				--type 'SampleData[SequencesWithQuality]' \
				--input-path ${manifest} \
				--output-path demux.qza \
				--input-format $input_format
			"""
```

File: main.nf [hunk 2]
```diff
@@ -791,7 +870,7 @@ process dada_trunc_parameter {
 	!params.untilQ2import
 
 	script:
-	if( !params.trunclenf || !params.trunclenr ){
+	if( params.trunclenf == false || params.trunclenr == false ){
```
Comment (d4straub) [inline main.nf:873]: Why this change? This seems inconsistent with the rest of the pipeline.

File: main.nf [hunk 3]
```diff
@@ -918,6 +1021,80 @@ if (!params.multipleSequencingRuns){
...
+		file("sequences.fasta")
+		file("dada_report.txt")
+
+		when:
+		!params.untilQ2import
+
+		script:
+		"""
+		# Quality filtering with DADA2 filterAndTrim
+		# Might want to add params.minLen and params.maxLen in the future
```
Comment (d4straub) [inline main.nf:1058]: Totally agreed, this should be in parameters, could you add this? Just keep the values you added here as default.

**Review by d4straub — CHANGES_REQUESTED** [review]
Looks great!
I like that you mimic Illumina output. That of course explains why there are basically no changes to `output.md`.

But I think what is additionally needed:
- update `CHANGELOG.md`
- test datasets, for (1) PacBio with SILVA (2) PacBio with UNITE and (3) Illumina with UNITE. At least one of these must be added in this PR, in my opinion.
- a discussion if --single_end is needed. I do not see the necessity and feel it is a unneeded parameter. But I am open to be convinced otherwise.

File: main.nf [hunk 4]
```diff
@@ -43,6 +44,7 @@ def helpMessage() {
 	                                create overlapping file names that crash MultiQC.
 	  --split [str]                 A string that will be used between the prepended run/folder name and the sample name. (default: "-")
 	                                May not be present in run/folder names and no underscore(s) allowed. Only used with "--multipleSequencingRuns"
+	  --pacbio			If PacBio data. Use this option together with --single_end and --manifest.
```
[the remark this answers is not in the document]
Comment (jtangrot) [reply main.nf:47]: Well, it was still present in the main.nf and some other places a couple of weeks ago, so I thought I should use it... It is not really needed as it is now, the pacbio option can be used to determine how to read input data, but it might be useful if other types of data is supported in the future? I have no strong opinions here.

[the remark this answers is not in the document]
Comment (d4straub) [reply main.nf:47]: Are some of the processes that are executed now with `params.single_end = true` also usable for single end Illumina data? If yes, than future additions (that I do not plan at this time) might profit. But I think in this case `single_end` should be set to true without specifying it. I dislike that currently `--single_end` and `--pacbio` need to be specified simultaneously. `--pacbio` defines that data is single end, there is no paired end pacbio data, is there?
If processes executed now with `params.single_end = true` are not potentially usable for single end Illumina data, I think it would be better to remove `--single_end` and use only `--pacbio`.

File: environment.yml [hunk 5]
```diff
@@ -65,7 +67,7 @@ dependencies:
 - click=7.0
 - cryptography=2.7
 - curl=7.65.3
-- cutadapt=2.6
+- cutadapt=2.8
```
[the remark this answers is not in the document]
Comment (jtangrot) [reply environment.yml:70]: Yes, at least version 2.8 is needed as the --revcomp option is used.

File: main.nf (hunk 2 above)
Comment (jtangrot) [reply main.nf:873]: This change is made to allow for trunclenr and trunclenf to be set to the value 0, which is interpreted in DADA2 as "no trunctaion". Needed for ITS sequences where we don't expect a given length.

Comment (d4straub) [reply main.nf:873]: Right, makes sense!

File: environment.yml (hunk 5 above)
[the remark this answers is not in the document]
Comment (d4straub) [reply environment.yml:70]: Ok, good to know. I guess we would next update QIIME2 to 2020.08 and [this contains 2.10](https://github.com/qiime2/environment-files/blob/master/2020.8/release/qiime2-2020.8-py36-linux-conda.yml), so actually no problem.

File: main.nf (hunk 3 above)
Comment (d4straub) [reply main.nf]: This seems resolved with the latest commit.

File: bin/make_dada2_report_pacbio.py [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+# the DADA2 denoising process, mimicking the report
+# generated by QIIME2.
+# Results are written to file "dada_report.tsv"
+#
+# Jeanette Tångrot
+
+#-- Import libraries, do initializations  --#
+import sys 
+
+file_out = "dada_report.txt"
```
[the remark this answers is not in the document]
Comment (jtangrot) [reply bin/make_dada2_report_pacbio.py]: Typo, fixed now. Thanks for noticing!

File: main.nf (hunk 4 above)
[the remark this answers is not in the document]
Comment (jtangrot) [reply main.nf]: I think the single end option is potentially usable for other types of single end data. However, it's a god idea to set it without specifying it for PacBio data. Thanks for the suggestion.

File: docs/output.md [hunk 7]
```diff
@@ -105,7 +105,7 @@ DADA2 reduces sequence errors and dereplicates sequences by quality filtering, d
 
 ### Taxonomic classification
```
[the remark this answers is not in the document]
Comment (jtangrot) [reply docs/output.md:112]: Yes, you're right, I should add something about UNITE.

[the remark this answers is not in the document]
Comment (jtangrot) [reply docs/output.md:112]: True, added a short line.

**Comment by jtangrot** [comment]
I think I have adressed all issues now. However, I might have failed to follow the instructions on uploading a test dataset and merged the files to the repo without doing a proper pull request (I did not think I had permissions to do such a mistake...).

**Comment by d4straub** [comment]
Great! I will have a detailed look today. I am actually also surprised that the test datasets are not protected branches. 

One issue I saw right now: While you added PacBio tests, they are actually not automatically executed in your PR.
Could you append in file `.github/workflows/ci.yml` the lines
```
      - name: Run pipeline with ITS PacBio reads and Unite database
        run: | 
          nextflow run ${GITHUB_WORKSPACE} -profile test_pacbio_its,docker
```

**Comment by d4straub** [comment]
Looks good!

**Review by d4straub — APPROVED** [review]
